modules/effects/arrow-effects-extensions/src/main/kotlin/arrow/effects/extensions/io.kt

```diff
@@ -143,18 +144,27 @@ interface IOConcurrent : Concurrent<ForIO>, IOAsync {
 
 }
 
+fun IO.Companion.concurrent(dispatchers: Dispatchers<ForIO>): Concurrent<ForIO> = object : IOConcurrent {
+  override fun dispatchers(): Dispatchers<ForIO> = dispatchers
+}
+
 @extension
-interface IOEffect: Effect<ForIO>, IOAsync {
+interface IOEffect : Effect<ForIO>, IOAsync {
   override fun <A> IOOf<A>.runAsync(cb: (Either<Throwable, A>) -> IOOf<Unit>): IO<Unit> =
     fix().runAsync(cb)
 }
 
-@extension
-interface IOConcurrentEffect: ConcurrentEffect<ForIO>, IOEffect, IOConcurrent {
+// FIXME default @extension are temporarily declared in arrow-effects-io-extensions due to multiplatform needs
```

Comment:
What is the actual limitation why `@extension` can't be used here?

Reply:
There's no value for Dispatchers in this module, as it's not JVM-dependent. This is here so users can do `IO.concurrent(MyDispatcher)` where they plug the kx.coroutines, reactor, or the rx ones.

Reply:
@pakoito what is to fix here?